python/tvm/contrib/binutil.py

```diff
@@ -125,74 +146,73 @@ def tvm_callback_relocate_binary(
     rel_bin : bytearray
         the relocated binary
     """
-    tmp_dir = util.tempdir()
-    rel_obj_path = tmp_dir.relpath("relocated.o")
-    ld_script_contents = ""
+    stack_pointer_init = stack_end - word_size
+
+    ld_script_contents = ''
     # TODO(weberlo): There should be a better way to configure this for different archs.
-    if "riscv" in toolchain_prefix:
-        ld_script_contents += "OUTPUT_ARCH( \"riscv\" )\n\n"
-    # TODO(weberlo): Generate the script in a more procedural manner.
-    ld_script_contents += """
+    if 'riscv' in toolchain_prefix:
+        ld_script_contents += 'OUTPUT_ARCH( \"riscv\" )\n\n'
+    ld_script_contents += f"""
+/* linker symbol for use in UTVMInit */
```

Comment:
consider define this piece of code in as a global variable, so that the code structure is more clear